pkg/subscriber/objectbucket/objectbucket_subscriber_item.go

```diff
@@ -216,16 +216,15 @@ func (obsi *SubscriberItem) doSubscription() error {
 
 		err = obsi.synchronizer.AddTemplates(syncsource, hostkey, []kubesynchronizer.DplUnit{{Dpl: dpltosync, Gvk: *validgvk}})
 		if err != nil {
-			klog.Info("eror in registering :", err)
-			err = utils.SetInClusterPackageStatus(&(obsi.Subscription.Status), dpltosync.GetName(), err, nil)
+			klog.Error("eror in registering :", err)
 
-			if err != nil {
-				klog.Info("error in setting in cluster package status :", err)
+			if serr := utils.SetInClusterPackageStatus(&(obsi.Subscription.Status), dpltosync.GetName(), err, nil); serr != nil {
+				klog.Error("error in setting in cluster package status :", err)
 			}
 
 			pkgMap[dpltosync.GetName()] = true
 
-			continue
+			return err
```

Comment:
This is a semi big change behaviour no? Before we would continue doing the others in the loop. 
Is it possible to keep our old behaviour?

Reply:
we could but I feel we should let the subscriber fail earlier, and operate in an atomic manner. 

the old logic is trying to do everything at once, if we go a full cache channel on the 5th item, I doubt we will be applied to apply 6th item right away.